### PR TITLE
Update and improve API documentation

### PR DESCRIPTION
**Pull request by galpeter**
A few parts of the API documentation was not clear enough.
Added extra details and examples.

### CODE REVIEW

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -2359,6 +2561,13 @@ jerry_get_utf8_string_size (const jerry_value_t value);
 
 Get the length of a string. Returns zero, if the value parameter is not a string.
 
+
+*Note:*
+- The difference from [jerry_get_string_size](#jerry_get_string_size) is that it
+  returns the number of bytes used for the string.
+- This is **not** the number of bytes required to store the string.
+
+
```
Comment (rerobika) [inline docs/02.API-REFERENCE.md]: IMO 1 new line is enough here.

**Comment by galpeter** [comment]
@rerobika I've did a bit of update (around the `jerry_string_to_char_buffer` method)

**Comment by rerobika** [comment]
Still LGTM.

File: docs/02.API-REFERENCE.md [hunk 2]
```diff
@@ -3440,6 +3695,17 @@ handler (const jerry_value_t function_obj,
 
   jerry_release_value (func_val);
   jerry_release_value (glob_obj);
+
+
```
Comment (dbatyai) [inline docs/02.API-REFERENCE.md]: Extra line here.

File: docs/02.API-REFERENCE.md [hunk 3]
```diff
@@ -4611,32 +4928,152 @@ jerry_define_own_property (const jerry_value_t obj_val,
...
 
+  // configure the property
   jerry_property_descriptor_t prop_desc;
   jerry_init_property_descriptor_fields (&prop_desc);
 
   jerry_value_t value_to_set;
 
-  ... // create or acquire value to set
+  // create or acquire value to set
+  // Fo example:
```
Comment (dbatyai) [inline docs/02.API-REFERENCE.md]: Fo**r** example

File: docs/02.API-REFERENCE.md [hunk 4]
```diff
@@ -5270,25 +5711,74 @@ jerry_foreach_object_property (jerry_value_t obj_val,
...
-  jerry_value_t object;
-  ... // receive or construct object
+  jerry_init (JERRY_INIT_EMPTY);
+
+  /* Construct an example object with a single property. */
+  jerry_value_t object = jerry_create_object ();
+  {
+    jerry_value_t test_property = jerry_create_string ((const jerry_char_t *) "DemoProp");
+    jerry_value_t test_value = jerry_create_number (3);
+    /* By default all properties added to an object is enumerable. */
```
Comment (dbatyai) [inline docs/02.API-REFERENCE.md]: is -> are

**Pull request merged**